src/averageAges.js

```diff
@@ -55,7 +76,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childMotherPairs = people.map((child) => {
+    const mother = (!onlyWithSon || child.sex === 'm')
+      && child.mother
+      && people.find((person) => (
+        person.name === child.mother)
+      );
+
+    return mother && [child.born, mother.born];
+  });
+
+  const ageDiff = childMotherPairs
+    .filter(x => x)
```

Comment:
I don't understand what is 'x', and 'pair[0]' and 'pair[1]'. Please use the correct names for variables.